content/Engineering/Programming/Compilers/generation.md

```diff
@@ -23,20 +29,24 @@ end while
 print array[0]
 ```
 
-This would be broken into 3 blocks:
-
-```text
-start
- |
- |    --------------------------------------------------------------
- |   /                                                              \
- v  v         true                                                   \
-[ n < 10 ] ----> [ x = read integer from file; array[n] = x; n = n + 1]
-  \
-   \  false
-    ----------> [ print array[0] ] --> end
-
-```
+This would be broken into 3 blocks. Note, all the logic in the loop can be thought of as a single block, as the operations are consecutive without any conditional logic (as long as me assume "read integer from file" can't fail)
+
```

Comment:
A small nit: the "block" in this case is the entire contents of the loop body.

I think your diagram is much better, but perhaps call it "loop block"? Or maybe have "block" shading around the other pieces to distinguish it from a statement?

Reply:
Ah, now I see it in the text above. That's good enough.